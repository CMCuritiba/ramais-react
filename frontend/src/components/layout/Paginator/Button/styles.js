import styled from 'styled-components';
import { darken } from 'polished';

export const Container = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  background-color: ${props =>
    props.atual ? darken(0.1, '#30666D') : '#30666D'};
  color: #fff;
  border: 0;
  border-radius: 4px;
  padding: 10px 15px;
  margin-left: 4px;
  transition: background 0.2s;

  &:hover {
    background: ${darken(0.1, '#30666D')};
  }

  &:disabled {
    cursor: not-allowed;
  }
`;
