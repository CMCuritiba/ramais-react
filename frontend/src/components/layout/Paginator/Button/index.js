import React from 'react';

import { Container } from './styles';

const Button = ({ value, handleClickPagina, page, loading }) => {
  return page === value ? (
    <Container
      onClick={() => handleClickPagina(value)}
      atual={1}
      disabled={loading || page === value}
    >
      {value}
    </Container>
  ) : (
    <Container
      onClick={() => handleClickPagina(value)}
      disabled={loading || page === value}
    >
      {value}
    </Container>
  );
};

export default Button;
