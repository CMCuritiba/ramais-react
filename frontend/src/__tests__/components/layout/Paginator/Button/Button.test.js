import React from 'react';

import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
  getByText,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from '../../../../../components/layout/Paginator/Button';

afterEach(cleanup);

describe('Button', () => {
  it('renderiza corretamente', () => {
    function clickPagina() {}

    const value = 1;
    const page = 1;
    const loading = false;

    const { container } = render(
      <Button
        value={value}
        page={page}
        loading={loading}
        handleClickPagina={clickPagina}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('testa click no botão', () => {
    function clickPagina(e) {
      expect(e).toBe(222);
    }

    const value = 222;
    const page = 1;
    const loading = false;

    const { container, getByText } = render(
      <Button
        value={value}
        page={page}
        loading={loading}
        handleClickPagina={clickPagina}
      />
    );

    const botao = getByText('222');

    fireEvent.click(botao);
  });

  it('testa botão desabilitado se mesma página', () => {
    const value = 222;
    const page = 222;
    const loading = false;

    const { container, getByText } = render(
      <Button value={value} page={page} loading={loading} />
    );

    const botao = getByText('222');

    expect(botao).toHaveAttribute('disabled');
  });

  it('testa botão desabilitado se carregando dados', () => {
    const value = 222;
    const page = 1;
    const loading = true;

    const { container, getByText } = render(
      <Button value={value} page={page} loading={loading} />
    );

    const botao = getByText('222');

    expect(botao).toHaveAttribute('disabled');
  });
});
