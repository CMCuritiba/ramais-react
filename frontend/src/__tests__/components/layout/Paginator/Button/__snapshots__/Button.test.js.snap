// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renderiza corretamente 1`] = `
<div>
  <button
    class="sc-bdVaJa hlskzD"
    disabled=""
  >
    1
  </button>
</div>
`;
